packages/web-api-scan-runner/src/runner/runner.ts

```diff
@@ -118,6 +121,64 @@ export class Runner {
         this.logger.logInfo('Page scan task completed.');
     }
 
+    private async updateWebsiteScanResultWithDiscoveredUrls(
+        pageScanResult: OnDemandPageScanResult,
+        newlyDiscoveredUrls: string[],
+    ): Promise<void> {
+        await this.retryHelper.executeWithRetries(
+            async () => this.updateWebsiteScanResultWithDiscoveredUrlsImpl(pageScanResult, newlyDiscoveredUrls),
+            async (error: Error) => {
+                this.logger.logError(`Failure to generate combined scan result. Retrying on error.`, {
+                    error: System.serializeError(error),
+                });
+            },
+            this.maxCombinedResultProcessingRetryCount,
+            1000,
+        );
+    }
+
+    private async updateWebsiteScanResultWithDiscoveredUrlsImpl(
```

Comment:
Can this logic be abstracted out into a separate file/class?

Reply:
Yeah, I think this is similar to the report case. I'm happy to do that in this PR but wasn't sure if that's the goal right now.